$OpenBSD: patch-lib_ruby_shared_rubygems_package_rb,v 1.1 2013/10/15 02:36:22 jeremy Exp $

Fix for rubygems not installing gem files starting with ".". rubygems
bug #680.

--- lib/ruby/shared/rubygems/package.rb.orig	Sat Oct  5 13:58:22 2013
+++ lib/ruby/shared/rubygems/package.rb	Thu Oct 10 10:42:10 2013
@@ -339,13 +339,9 @@ EOM
   def extract_tar_gz io, destination_dir, pattern = "*" # :nodoc:
     open_tar_gz io do |tar|
       tar.each do |entry|
-        # Some entries start with "./" which fnmatch does not like, see github
-        # issue #644
-        full_name = entry.full_name.sub %r%\A\./%, ''
+        next unless File.fnmatch pattern, entry.full_name, File::FNM_DOTMATCH
 
-        next unless File.fnmatch pattern, full_name
-
-        destination = install_location full_name, destination_dir
+        destination = install_location entry.full_name, destination_dir
 
         FileUtils.rm_rf destination
 
